chain/chain/src/state_transition_data.rs

```diff
@@ -0,0 +1,265 @@
+use std::collections::hash_map::Entry;
+
+use std::collections::HashMap;
+
+use borsh::{BorshDeserialize, BorshSerialize};
+use near_chain_primitives::error::Error;
+use near_primitives::block::Block;
+use near_primitives::checked_feature;
+use near_primitives::hash::CryptoHash;
+use near_primitives::types::{BlockHeight, ShardId};
+use near_primitives::utils::{get_block_shard_id, get_block_shard_id_rev};
+use near_store::db::STATE_TRANSITIONS_METADATA_KEY;
+use near_store::{DBCol, StorageError};
+use once_cell::unsync::OnceCell;
+
+use crate::{Chain, ChainStore, ChainStoreAccess};
+
+#[derive(BorshSerialize, BorshDeserialize, PartialEq, Eq, Default, Debug)]
+struct StateTransitionsMetadata {
+    /// Represents max not-yet-GCed height for each shard
+    start_heights: HashMap<ShardId, BlockHeight>,
+}
+
+impl Chain {
+    pub(crate) fn garbage_collect_state_transition_data(&self, block: &Block) -> Result<(), Error> {
```

Comment:
Do we want to push this function to `chain/chain/src/garbage_collection.rs`?

I understand this isn't really GC code so we can keep it here as well.

Reply:
I initially considered putting it there, but decided against it for the following reasons:
1. as you've mentioned it is not a part of chain GC, so can be confusing
2. `garbage_collection.rs` file is already pretty big
3. it is more ergonomic if everything related to state transition data is in one file

Reply:
Another quick optional suggestion, how about we put this in the `chain/client/src/stateless_validation/....` directory?

Reply:
client is one level above chain, so it is not possible to have `impl Chain` as part of `chain/client`
I will move it to `chain/src/stateless_validation` instead. `chunk_endorsements.rs` also belongs there, will move it in a separate PR.